### DISCUSSION TITLE
Clear current backlog and queue new set of commands in one http request

### DISCUSSION THREAD

**Discussion by lemmy** [arendst/Tasmota#20331] [Q&A]
I need to clear the current backlog of commands using the `Backlog` command and then immediately queue a new set of commands, all within a single HTTP request. However, the two requests (here with unescaped spaces and semicolons for readability) I tried are not working:

`.../cm?cmnd=Backlog ; BackLog ; Power1 On`
`.../cm?cmnd=Backlog?cmnd=Backlog ; Power1 On`

**Comment by barbudor** [comment]
You can't use Backlog within Backlog
And your 2nd line doesn't make any sense with regards to HTTP

I'm not sure I understand why you think you need to do that
May be elaborating a bit more on what you are trying to achieve and what brought you to that path would be useful

**Reply by lemmy** [reply]
I want to clear all current commands that have been scheduled.  Consider this example:

First request `Backlog Power1 On;Delay 3000;Power1 Off` at time `N`.  A second request at `N+2000` shall clear the current delayed `Power1 Off` command that would fire in `1000` before scheduling, e.g., another delayed power off for a time `M > N+3000`.

**Reply by barbudor** [reply]
Backlog is not meant for what you are using it for. It's not reliable. Any other command with backlog or not may break the delay. 

Either use Pulsetime if you always have a timed-On or Off
Or use RuleTimer

**Reply by lemmy** [reply]
For documentation purposes, the command [`Backlog Power1 On; PulseTime1 N`](https://tasmota.github.io/docs/Commands/#control) nearly meets my needs, powering off automatically after a duration of 'N'. This duration can be reset with subsequent `Backlog Power1 On; PulseTime1 N`. However, a limitation is the necessity to use `PulseTime1 0` to clear the pulse state. Without this, any future `Power1 On` commands will continue to turn the device off after the previously set 'N' duration.  A workaround could be to always invoke `Backlog ; PulseTime1 0; Power1 On` instead of `Power1 On`.

Will experiment with `RuleTimer` next.

**Reply by barbudor** [reply]
what I meant but badly explained: if you always need to have the same power on duration, use Pulsetime and configure it once for all. A single 'power on' will start the timing. It will either end automatically or be cleared by a 'power off'
If you have different use cases, then a rule timer is preferable
Everytime you are using the pulsetime command, the new configuration value is written to flash. So you are wearing your flash quicker. 

When 'RuleTimerX nnnn' is used, it restarts the timer whether it is stopped or already running (re-trigger able timer)
If you want to cancel a running timer, issue 'RuleTimerX 0'
(X is the index of the timer)


Hope this helps

**Reply by sfromis** [reply]
Using `PulseTime`, you can extend the duration by doing a new `Power on`, and cut it short using a `Power off`, but it is indeed only intended as a configuration option you're not setting each time.
